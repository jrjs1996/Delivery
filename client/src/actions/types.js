export const FETCH_CUSTOMERS = 'FETCH_CUSTOMERS';
export const ADD_CUSTOMER = 'ADD_CUSTOMER';

export const ADD_ORDER = 'ADD_ORDER';
export const FETCH_ORDERS = 'FETCH_ORDER';
export const COMPLETE_ORDER = 'COMPLETE_ORDER';
export const GET_CUSTOMER = 'GET_CUSTOMER';

export const LOGIN_ADMIN = 'LOGIN_ADMIN';
export const FETCH_ADMINS = 'FETCH_ADMINS';

export const FETCH_MENU = 'FETCH_MENU';
export const ADD_MENU_ITEM = 'ADD_MENU_ITEM';
export const UPDATE_MENU_ITEM = 'UPDATE_MENU_ITEM';
export const DELETE_MENU_ITEM = 'DELETE_MENU_ITEM';
