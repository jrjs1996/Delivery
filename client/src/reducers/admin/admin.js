import {
  CREATE_ADMIN, FETCH_ADMINS, UPDATE_ADMIN, UPDATE_CURRENT_ADMIN, DELETE_CURRENT_ADMIN,
} from '../../actions/types';
import { createItems, updateItem, insertItem } from '../utils/utils';

const initialState = {
  admins: {},
  currentAdmin: {},
};

export default function (state = initialState, action) {
  switch (action.type) {
    case CREATE_ADMIN:
      return {
        ...state,
        admins: insertItem(state.admins, action.payload, '_id'),
      };
    case DELETE_CURRENT_ADMIN:
      return {
        ...state,
        currentAdmin: {},
      };
    case FETCH_ADMINS:
      return {
        ...state,
        admins: createItems(action.payload, '_id'),
      };
    case UPDATE_ADMIN:
      return {
        ...state,
        admins: updateItem(state.admins, action.payload, '_id'),
      };
    case UPDATE_CURRENT_ADMIN:
      return {
        ...state,
        currentAdmin: action.payload,
      };
    default:
      return state;
  }
}
