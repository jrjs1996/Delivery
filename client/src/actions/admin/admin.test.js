import configureStore from 'redux-mock-store';
import * as adminActions from './admin';
import * as actionTypes from '../types';
import mockAxios from '../../../__mocks__/axios';

const mockStore = configureStore();
const store = mockStore();

describe('Create admin', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches correct info when successful', async () => {
    mockAxios.post.mockImplementationOnce((url, postData) => Promise.resolve({
      data: { username: postData.username },
    }));

    await adminActions.createAdmin({
      username: 'TestUser',
      password: 'TestPassword',
    })(store.dispatch);

    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches error when unsuccessful', async () => {
    mockAxios.giveError(mockAxios.post);
    await adminActions.createAdmin({
      username: 'TestUser',
      password: 'TestPassword',
    })(store.dispatch);

    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('Fetch Admins', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches correct info when successful (Payload is res data)', async () => {
    await adminActions.fetchAdmins()(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches error when unsuccessful', async () => {
    mockAxios.giveError(mockAxios.get);
    await adminActions.fetchAdmins()(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('Get current admin info', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches correct info when successful (Payload is res data)', async () => {
    await adminActions.getCurrentAdminInfo()(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches error when unsuccessful', async () => {
    mockAxios.giveError(mockAxios.get);
    await adminActions.getCurrentAdminInfo()(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('Login', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches correct info when successful (Payload is provided username)', async () => {
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ status: 200, data: { mock: 'MockData', token: 'MockToken' } });
    });
    await adminActions.login({ username: 'TestUsername', password: 'TestPassword' })(store.dispatch);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches error when unsuccessful', async () => {
    mockAxios.giveError(mockAxios.post);
    await adminActions.login({
      username: 'TestUser',
      password: 'TestPassword',
    })(store.dispatch);

    expect(store.getActions()).toMatchSnapshot();
  });
});


describe('Update current admin', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches correct info when successful (Payload is res data)', async () => {
    await adminActions.updateCurrentAdmin({
      username: 'TestUser',
      password: 'TestPassword',
    })(store.dispatch);

    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches error when unsuccessful', async () => {
    mockAxios.giveError(mockAxios.put);
    await adminActions.updateCurrentAdmin({
      username: 'TestUser',
      password: 'TestPassword',
    })(store.dispatch);

    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('updateAdmin', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('Dispatches correct info when successful (Payload is submitted data)', async () => {
    await adminActions.updateAdmin({
      username: 'TestUser',
      password: 'TestPassword',
    })(store.dispatch);

    expect(store.getActions()).toMatchSnapshot();
  });

  it('Dispatches error when unsuccessful', async () => {
    mockAxios.giveError(mockAxios.put);
    await adminActions.updateAdmin({
      username: 'TestUser',
      password: 'TestPassword',
    })(store.dispatch);

    expect(store.getActions()).toMatchSnapshot();
  });
});
