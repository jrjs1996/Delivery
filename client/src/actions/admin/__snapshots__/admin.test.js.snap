// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create admin Dispatches correct info when successful 1`] = `
Array [
  Object {
    "payload": Object {
      "username": "TestUser",
    },
    "type": "CREATE_ADMIN",
  },
]
`;

exports[`Create admin Dispatches error when unsuccessful 1`] = `
Array [
  Object {
    "payload": "Unable to add admin",
    "type": "ERROR",
  },
]
`;

exports[`Fetch Admins Dispatches correct info when successful (Payload is res data) 1`] = `
Array [
  Object {
    "payload": Object {
      "mock": "MockData",
    },
    "type": "FETCH_ADMINS",
  },
]
`;

exports[`Fetch Admins Dispatches error when unsuccessful 1`] = `
Array [
  Object {
    "payload": "Unable to fetch admins",
    "type": "ERROR",
  },
]
`;

exports[`Get current admin info Dispatches correct info when successful (Payload is res data) 1`] = `
Array [
  Object {
    "payload": Object {
      "mock": "MockData",
    },
    "type": "UPDATE_CURRENT_ADMIN",
  },
]
`;

exports[`Get current admin info Dispatches error when unsuccessful 1`] = `
Array [
  Object {
    "payload": "Unable to get current admin info.",
    "type": "ERROR",
  },
]
`;

exports[`Login Dispatches correct info when successful (Payload is provided username) 1`] = `
Array [
  Object {
    "payload": Object {
      "mock": "MockData",
      "token": "MockToken",
    },
    "type": "UPDATE_CURRENT_ADMIN",
  },
]
`;

exports[`Login Dispatches error when unsuccessful 1`] = `
Array [
  Object {
    "payload": "Unable to log in",
    "type": "ERROR",
  },
]
`;

exports[`Update current admin Dispatches correct info when successful (Payload is res data) 1`] = `
Array [
  Object {
    "payload": Object {
      "mock": "MockData",
    },
    "type": "UPDATE_CURRENT_ADMIN",
  },
]
`;

exports[`Update current admin Dispatches error when unsuccessful 1`] = `
Array [
  Object {
    "payload": "Unable to update current admin.",
    "type": "ERROR",
  },
]
`;

exports[`updateAdmin Dispatches correct info when successful (Payload is submitted data) 1`] = `
Array [
  Object {
    "payload": Object {
      "password": "TestPassword",
      "username": "TestUser",
    },
    "type": "UPDATE_ADMIN",
  },
]
`;

exports[`updateAdmin Dispatches error when unsuccessful 1`] = `
Array [
  Object {
    "payload": "Unable to update admin.",
    "type": "ERROR",
  },
]
`;
